pkgs/by-name/ze/zed-editor/package.nix

```diff
@@ -0,0 +1,220 @@
+        Metal
+        Security
+        SystemConfiguration
+        VideoToolbox
+      ]
+    );
+
+  buildFeatures = [ "gpui/runtime_shaders" ];
+
+  env = {
+    ZSTD_SYS_USE_PKG_CONFIG = true;
+    FONTCONFIG_FILE = makeFontsConf {
+      fontDirectories = [
+        "${src}/assets/fonts/zed-mono"
+        "${src}/assets/fonts/zed-sans"
+      ];
+    };
+  };
+
+  postFixup = lib.optionalString stdenv.isLinux ''
+    patchelf --add-rpath ${vulkan-loader}/lib $out/bin/*
+    patchelf --add-rpath ${wayland}/lib $out/bin/*
+  '';
+
+  # Fail with "unimplemented error"
+  checkFlags = lib.optionals stdenv.hostPlatform.isLinux [
+    "--skip=test_base_keymap"
+    "--skip=test_bundled_settings_and_themes"
+    "--skip=test_disabled_keymap_binding"
+    "--skip=test_navigation"
+    "--skip=test_new_empty_workspace"
+    "--skip=test_open_and_save_new_file"
+    "--skip=test_open_entry"
+    "--skip=test_opening_excluded_paths"
+    "--skip=test_open_paths"
+    "--skip=test_open_paths_action"
+    "--skip=test_pane_actions"
+    "--skip=test_reopening_closed_items"
+    "--skip=test_save_conflicting_item"
+    "--skip=test_setting_language_when_saving_as_single_file_worktree"
+    "--skip=test_window_edit_state"
+  ];
+
+  postInstall = ''
+    mv $out/bin/Zed $out/bin/zed-stable
+    ln -s $out/bin/zed-stable $out/bin/zed
+    install -D ${src}/crates/zed/resources/app-icon@2x.png $out/share/icons/hicolor/1024x1024@2x/apps/Zed.png
+    install -D ${src}/crates/zed/resources/app-icon.png $out/share/icons/hicolor/512x512/apps/Zed.png
+    cp -r ${
+      makeDesktopItem {
+        name = "dev.zed.Zed";
+        exec = "zed-stable %F";
+        tryExec = "zed-stable";
+        icon = "Zed";
+        comment = meta.description;
+        desktopName = "Zed";
+        genericName = "Text Editor";
+        categories = [
+          "Utility"
+          "TextEditor"
+          "Development"
+        ];
+        keywords = [
+          "Text"
+          "Editor"
+        ];
+        terminal = false;
+        type = "Application";
+        mimeTypes = [
+          "inode/directory"
+          "text/plain"
+          "text/x-makefile"
+          "text/x-c++hdr"
+          "text/x-c++src"
+          "text/x-chdr"
+          "text/x-csrc"
+          "text/x-java"
+          "text/x-moc"
+          "text/x-pascal"
+          "text/x-tcl"
+          "text/x-tex"
+          "application/x-shellscript"
+          "text/x-c"
+          "text/x-c++"
+        ];
+      }
+    }/* $out
+    chmod 0555 $out/share/applications/*.desktop
+  '';
+
+  meta = with lib; {
+    description = "A high-performance, multiplayer code editor from the creators of Atom and Tree-sitter";
+    homepage = "https://zed.dev";
+    changelog = "https://github.com/zed-industries/zed/releases/tag/v${version}";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [
+      GaetanLepage
+      niklaskorz
+    ];
+    mainProgram = "zed";
+    platforms = platforms.all;
```

Comment:
buildRustPackage sets this to unix by default which IMO is probably more correct.

```suggestion
```

Reply:
Not so sure, upstream Windows support is *very* quickly catching up with Linux and Darwin is the primary upstream target? Unless there are some nixpkgs-specific things limiting this.